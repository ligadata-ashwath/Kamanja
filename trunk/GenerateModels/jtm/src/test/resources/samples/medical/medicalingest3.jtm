{
  "header": {
    "namespace": "com.ligadata.models.samples.medical3",
    "version": "0.0.1",
    "language" : "scala",
    "minVersion" : "2.11",
    "description": "Finance load transaction message",
    "name": "TransactionIngest"
  },
  "imports": {
    "packages": [ "com.ligadata.runtime.Conversion"]
  },
  "aliases": {
    "messages": {
      "m1": "com.ligadata.kamanja.samples.messages.TransactionMsgIn",
      "o2": "com.ligadata.kamanja.samples.messages.HL71"
    }
  },
  "transformations": {
    "transactionmsg": {
      "dependsOn": [ ["m1"] ],
      "computes": {
        "arraydata":{ "type": "Array[String]", "val": "$data.split(\",\")", "comment": "Split the incoming data" },
        "typeName": { "type": "String", "val": "$arraydata(0)", "comment": "extract the type" }
      },
      "outputs": {

        "o2": {
          "where": "\"com.ligadata.kamanja.samples.messages.HL7\" == $typeName",
          "mapbyposition": {
            "arraydata" : [ "-", "desynpuf_id", "clm_id", "clm_from_dt", "clm_thru_dt"]
          }
        }
      }
    }
  }
}
