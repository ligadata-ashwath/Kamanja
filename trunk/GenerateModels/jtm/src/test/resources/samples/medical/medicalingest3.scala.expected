/*
* Copyright 2016 ligaDATA
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*     http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/
package com.ligadata.models.samples.finance.V1
import com.ligadata.KamanjaBase._
import com.ligadata.KvBase.TimeRange
import com.ligadata.kamanja.metadata.ModelDef
import com.ligadata.runtime.Log
import com.ligadata.runtime.Conversion
class TransactionIngestFactory(modelDef: ModelDef, nodeContext: NodeContext) extends ModelInstanceFactory(modelDef, nodeContext) {
  override def createModelInstance(): ModelInstance = return new TransactionIngest(this)
  override def getModelName: String = "com.ligadata.models.samples.finance.TransactionIngest"
  override def getVersion: String = "0.0.1"
  override def createResultObject(): ModelResultBase = new MappedModelResults()
}
class TransactionIngest(factory: ModelInstanceFactory) extends ModelInstance(factory) {
  val conversion = new com.ligadata.runtime.Conversion
  val log = new com.ligadata.runtime.Log(this.getClass.getName)
  import log._
  override def execute(txnCtxt: TransactionContext, execMsgsSet: Array[ContainerOrConcept], triggerdSetIndex: Int, outputDefault: Boolean): Array[ContainerOrConcept] = {
    if (isTraceEnabled)
    Trace(s"Model::execute transid=%d triggeredset=%d outputdefault=%s".format(txnCtxt.transId, triggerdSetIndex, outputDefault.toString))
    if(isDebugEnabled)
    {
      execMsgsSet.foreach(m => Debug( s"Input: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    //
    //
    def exeGenerated_transactionmsg_1(msg1: com.ligadata.kamanja.samples.messages.V1000000.TransactionMsgIn): Array[MessageInterface] = {
      Debug("exeGenerated_transactionmsg_1")
      // Split the incoming data
      val arraydata: Array[String] = msg1.data.split(",")
      // extract the type
      val typeName: String = arraydata(0)
      def process_o2(): Array[MessageInterface] = {
        Debug("exeGenerated_transactionmsg_1::process_o2")
        if (!("com.ligadata.kamanja.samples.messages.HL7" == typeName)) {
          Debug("Filtered: transactionmsg@o2")
          return Array.empty[MessageInterface]
        }
        val result = com.ligadata.kamanja.samples.messages.V1000000.HL71.createInstance
        result.desynpuf_id = arraydata(1)
        result.clm_id = conversion.ToLong(arraydata(2))
        result.set("clm_from_dt", arraydata(3))
        result.set("clm_thru_dt", arraydata(4))
        Array(result)
      }
      process_o2()
    }
    // Evaluate messages
    val msgs = execMsgsSet.map(m => m.getFullTypeName -> m).toMap
    val msg1 = msgs.getOrElse("com.ligadata.kamanja.samples.messages.TransactionMsgIn", null).asInstanceOf[com.ligadata.kamanja.samples.messages.V1000000.TransactionMsgIn]
    // Main dependency -> execution check
    //
    val results: Array[MessageInterface] =
    (if(msg1!=null) {
      exeGenerated_transactionmsg_1(msg1)
    } else {
      Array.empty[MessageInterface]
    }) ++
    Array.empty[MessageInterface]
    if(isDebugEnabled)
    {
      results.foreach(m => Debug( s"Output: %s -> %s".format(m.getFullTypeName, m.toString())))
    }
    results.asInstanceOf[Array[ContainerOrConcept]]
  }
}
